numpy/ma/core.py

```diff
@@ -6535,9 +6530,7 @@ def compressed(x):
         Equivalent method.
 
     """
-    if not isinstance(x, MaskedArray):
-        x = asanyarray(x)
-    return x.compressed()
+    return asanyarray(x).compressed()
```

Comment:
Yes, better in another PR. As is, this is a bug fix, but that would be maintenance.